src/models/transformer.h

```diff
@@ -309,7 +310,8 @@ class Transformer : public EncoderOrDecoderBase {
         && cache_[prefix + "_values"]->shape().elements() == values->shape().elements()) {
       vh = cache_[prefix + "_values"];
     } else {
-      auto Wv = graph_->param(prefix + "_Wv", {dimModel, dimModel}, inits::glorotUniform(true, true, depthScaling_ ? 1.f / sqrtf((float)depth_) : 1.f));
+      int dimValues = values->shape()[-1];
```

Comment:
As above, please add a comment when this will be different from `dimModel`.

Reply:
done